src/SchedulerSubsystem/Scheduler.java

```diff
@@ -51,30 +60,60 @@ private boolean isElevatorStopping(ElevatorStateEvent e) {
 
         // Create new object for union set to avoid funny business of destination events
         // being added to the scheduler's floor requests!
-        Set<DestinationEvent> union = new HashSet<DestinationEvent>();
-        union.addAll(floorRequests);
+        Set<DestinationEvent> union = new HashSet<>();
+        union.addAll(floorRequestsToTime.keySet());
         union.addAll(e.passengerCountMap().keySet());
 
         return union.contains(new DestinationEvent(e.currentFloor(), getElevatorDirection(e)));
     }
 
     /**
      * Gets the direction an elevator is travelling.
+     * Precondition: At least one passenger in the elevator or at least one floor request in floorRequestsToTime.
      *
-     * @param e Elevator state to get direction from.
+     * @param event Elevator state to get direction from.
      * @return Direction elevator is travelling.
      */
-    private Direction getElevatorDirection(ElevatorStateEvent e) {
-        // TODO(@braeden): should always return a valid direction or throw error
-        //   I'm worried about returning a null value!
-        // TODO(@braeden): should also assert that every direction is same?
-        //   see Elevator.getDirection
+    private Direction getElevatorDirection(ElevatorStateEvent event) {
+        // Find direction in elevator if elevator has passengers.
         Direction direction = null;
-        for (DestinationEvent event: e.passengerCountMap().keySet()) {
-            direction = event.direction();
+        for (DestinationEvent e : event.passengerCountMap().keySet()) {
+            if (direction == null) {
+                direction = e.direction();
+            }
+            if (direction != e.direction()) {
+                throw new RuntimeException("Mismatched passenger direction in elevator");
+            }
+        }
+        // Find direction from oldest floor request.
+        if (direction == null) {
+            direction = getOldestFloorDirFromElevator(event.currentFloor());
+        }
+        // No passengers on elevator and no floor requests. (Goes against precondition)
+        if (direction == null) {
+            throw new RuntimeException("No passenger on elevator and no floor requests");
         }
         return direction;
-    };
+    }
+
+    /**
+     * Returns the direction of the oldest floor request based on the elevator's current floor.
+     *
+     * @param currentFloor The floor number the elevator is currently on.
+     * @return Direction of oldest floor request relative to currentFloor, null if there are no floor requests.
+     */
+    private Direction getOldestFloorDirFromElevator(int currentFloor) {
+        if (floorRequestsToTime.isEmpty()) return null;
+        Long waitTime = (long) -1;
```

Comment:
I think there might be a way to get a MAX LONG CONSTANT type thing? Should use that